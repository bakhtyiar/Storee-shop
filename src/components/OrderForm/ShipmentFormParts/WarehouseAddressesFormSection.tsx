import React from 'react';
import {warehouses} from "../../../utils/constants";
import {Form} from "react-bootstrap";
import {Field} from "formik";

const WarehouseAddressesFormSection = ({
    as,
    name,
    notInline,
    type,
    label
}: any) => {
    return (
        <Form.Group data-testid="warehouse-addresses-form-section">
            
            <h6>{label || 'Warehouse address'}</h6>
            {
                warehouses.map((warehouse, index) => (
                    <Form.Check
                        as={as || Field}
                        inline={!notInline}
                        name={name || "address"}
                        label={warehouse.address}
                        value={warehouse.address}
                        type={type || 'radio'}
                        id={`warehouse-${index}`}
                        key={`warehouse-${index}`}
                    />
                ))
            }
        </Form.Group>
    );
};

export default WarehouseAddressesFormSection;