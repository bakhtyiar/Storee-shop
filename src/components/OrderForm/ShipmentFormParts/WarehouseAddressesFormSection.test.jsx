import {screen} from "@testing-library/react";
import {renderWithFormikContext} from "../../../../tests/integration-helpers/renderWithFormikContext";
import WarehouseAddressesFormSection from "./WarehouseAddressesFormSection";

describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithFormikContext(<WarehouseAddressesFormSection/>);
        expect(screen.getByTestId("warehouse-addresses-form-section")).toMatchSnapshot();
    })
})