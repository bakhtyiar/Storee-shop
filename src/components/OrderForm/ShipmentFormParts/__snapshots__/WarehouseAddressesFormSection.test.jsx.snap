// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<div
  data-testid="warehouse-addresses-form-section"
>
  <h6>
    Warehouse address
  </h6>
  <div
    class="form-check form-check-inline"
  >
    <input
      class="form-check-input"
      id="warehouse-0"
      name="address"
      type="radio"
      value="California Main st. 11"
    />
    <label
      class="form-check-label"
      for="warehouse-0"
      title=""
    >
      California Main st. 11
    </label>
  </div>
  <div
    class="form-check form-check-inline"
  >
    <input
      class="form-check-input"
      id="warehouse-1"
      name="address"
      type="radio"
      value="Kansas Main st. 25"
    />
    <label
      class="form-check-label"
      for="warehouse-1"
      title=""
    >
      Kansas Main st. 25
    </label>
  </div>
  <div
    class="form-check form-check-inline"
  >
    <input
      class="form-check-input"
      id="warehouse-2"
      name="address"
      type="radio"
      value="Ohio Main st. 7"
    />
    <label
      class="form-check-label"
      for="warehouse-2"
      title=""
    >
      Ohio Main st. 7
    </label>
  </div>
  <div
    class="form-check form-check-inline"
  >
    <input
      class="form-check-input"
      id="warehouse-3"
      name="address"
      type="radio"
      value="Washington Main st. 2"
    />
    <label
      class="form-check-label"
      for="warehouse-3"
      title=""
    >
      Washington Main st. 2
    </label>
  </div>
  <div
    class="form-check form-check-inline"
  >
    <input
      class="form-check-input"
      id="warehouse-4"
      name="address"
      type="radio"
      value="Louisiana Main st. 21"
    />
    <label
      class="form-check-label"
      for="warehouse-4"
      title=""
    >
      Louisiana Main st. 21
    </label>
  </div>
  <div
    class="form-check form-check-inline"
  >
    <input
      class="form-check-input"
      id="warehouse-5"
      name="address"
      type="radio"
      value="Hawaii Main st. 6"
    />
    <label
      class="form-check-label"
      for="warehouse-5"
      title=""
    >
      Hawaii Main st. 6
    </label>
  </div>
</div>
`;
